### DISCUSSION TITLE
Langfuse was not able to parse the LLM model

### DISCUSSION THREAD

**Discussion by fdbvega** [langfuse/langfuse#2379] [Support] [answered]
Hi, 
I'm having this issue:
"Langfuse was not able to parse the LLM model. The LLM call will be recorded without model name."

I'm tracing the call to the code but I'm not able to get the cost of the call

I'm using AWS Beadrock LLM

Does anybody know about it?

**Comment by marcklingen** [comment]
Can you add code to reproduce this warning? The SDKs parse out the model name from the Langchain class and you can create a PR to add support for the class that you use

**Reply by fdbvega** [reply]
Thank you fou supporting me. I'm ussing AWS Beadrock, so I would say Langchain class is not been used.
I can not post code but it's trivial.
I use a langfuse decorator above the LLM  function call

This procedure is working with ChatGPT y several LLMs but not with Beadrock and the call is the same

**Reply by marcklingen** [reply]
Thanks for the context! I’d appreciate it if you posted a code snippet including imports

**Reply by fdbvega** [reply]
Those are my imports

import json
import os
import boto3
import regex
from langchain.chains.summarize import load_summarize_chain
from langchain.llms.bedrock import Bedrock
from langchain_openai import OpenAI
from langchain.prompts import PromptTemplate
from langchain.text_splitter import RecursiveCharacterTextSplitter

from langfuse.openai import openai
from langfuse import Langfuse
from langfuse.callback import CallbackHandler
from langfuse.decorators import observe


@observe(as_type="generation")
def invoke_bedrock(LLM_kwargs):
    return Bedrock(**LLM_kwargs)


@observe()
def lambda_handler(event, context):
    llm = invoke_bedrock(LLM_kwargs)
    .
    .
    .
    return result

**Reply by hassiebp** [reply]
Thanks for sharing your repro example, @fdbvega! We have recently improved parsing support for Bedrock. Could you please upgrade to the latest version of our Langfuse-SDK and retry? If the issue persist, 
- what exact model_id are you requesting from Bedrock?
- what is the exact Langfuse and Langchain versions you are on?

**Reply by fdbvega** [reply]
Hi,
our Python versions are:
Langfuse == 2.33.0
Langchain == 0.2.1

Bedrock model:
"model_id": "anthropic.claude-v2"

**Reply by fdbvega** [reply]
We've update langfuse to 2.38.1 but we can not get tracking costs, it's set to zero

**Reply by hassiebp** [reply]
Could you please try using 
```python
from langchain_aws import BedrockLLM
```
as suggested in the Langchain docs? https://python.langchain.com/v0.2/docs/integrations/llms/bedrock/

**Reply by hassiebp** [reply]
@fdbvega Did you had the chance to try with `BedrockLLM`?

**Reply by fdbvega** [reply]
yes I did, but nothing new happened

**Reply by hassiebp** [reply, marked as answer]
Hi @fdbvega, 

There are a few things that might have gone wrong here. 
- Could you please share your full example including the BedrockLLM calls? 
- Do you see any Langfuse related errors in your console when executing your chain? 

This would be quite helpful as I am not able to reproduce the issue at the moment.

**Comment by marcklingen** [comment]
`from langchain.llms.bedrock import Bedrock`

you could add the model extraction logic to the Python sdk. Otherwise, happy to pick this up soon

**Reply by fdbvega** [reply]
This import was in the firts cell, so nothing new happens...
